### PR TITLE
Replace string templates with a query builder using SQLAlchemy

### PR DESCRIPTION
**Pull request by marksteve**
Issue: #7

_Currently depends on [TM's fork of pybigquery](https://github.com/thinkingmachines/pybigquery)_

### CODE REVIEW

File: geomancer/spells/base.py [hunk 1]
```diff
@@ -19,64 +19,123 @@
...
+
+        # Filter
+        query = select(
+            [
+                query,
+                func.row_number()
+                .over(
+                    partition_by=query.c["__index_level_0__"],
+                    # TODO: order_by should be a parameter
+                    order_by=query.c[self.feature_name].desc(),
```
Comment (ljvmiranda921) [inline geomancer/spells/base.py]: @marksteve  Will this work, say,  if we want the `NumberOf` POIs `within` a certain range? The original BQ query looks like this:

```sql
WITH
  pairs AS (
  WITH
    pois AS (
    SELECT
      osm_id,
      pois.WKT AS WKT
    FROM
      `osm.gis_osm_pois_free_1` AS pois)
  SELECT
    ST_DISTANCE(points.WKT,
      pois.WKT) AS dist,
    points.WKT AS points_WKT,
    points.pkey AS house_pkey,
    pois.WKT AS roads_WKT,
    pois.osm_id AS osm_id
  FROM
    `houses` AS points,
    pois AS pois
  WHERE
    ST_DISTANCE(points.WKT,
      pois.WKT) < 0.5*1000)
SELECT
  house_pkey,
  ST_ASTEXT(points_WKT) AS house_WKT,
  COUNT(DISTINCT(osm_id)) AS num_pois
FROM
  pairs
GROUP BY
  house_pkey,
  house_WKT
```

Comment (ljvmiranda921) [reply geomancer/spells/base.py]: If not, my suggestion is to put the whole query (including the filters) inside the `query()` method.

**Review by ljvmiranda921 — CHANGES_REQUESTED** [review]

File: geomancer/spells/base.py (hunk 1 above)
Comment (marksteve) [reply geomancer/spells/base.py]: Gets. Ok will modify

**Review by ljvmiranda921 — APPROVED** [review]
LGTM! Thank you @marksteve !

**Pull request merged**